cmd/account/mgmt/account-unassign.go

```diff
@@ -0,0 +1,405 @@
+	return &accountUnassignOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+func (o *accountUnassignOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+	if o.username == "" && o.accountID == "" {
+		return cmdutil.UsageErrorf(cmd, "Please provide either an username or account ID")
+	}
+	return nil
+}
+
+func (o *accountUnassignOptions) run() error {
+
+	var (
+		accountUsername string
+		accountIdList   []string
+		destinationOU   string
+		rootID          string
+	)
+
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	if o.payerAccount == "osd-staging-1" {
+		rootID = OSDStaging1RootID
+		destinationOU = OSDStaging1OuID
+	} else if o.payerAccount == "osd-staging-2" {
+		rootID = OSDStaging2RootID
+		destinationOU = OSDStaging2OuID
+	} else {
+		return fmt.Errorf("invalid payer account provided")
+	}
+
+	o.awsClient = awsClient
+
+	if o.accountID != "" {
+
+		err := o.checkForHiveNameTag(o.accountID)
+		if err != nil {
+			return err
+		}
+		accountIdList = append(accountIdList, o.accountID)
+
+	}
+
+	if o.username != "" {
+		// Check that username is not a hive
+		if strings.HasPrefix(o.username, "hive") {
+			return ErrHiveNameProvided
+		}
+
+		accountUsername = o.username
+		accountIdList, err = o.listAccountsFromUser(accountUsername)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Printf("Are you sure you want to unassign the accounts? [y/n] ")
+	reader := bufio.NewReader(os.Stdin)
+
+	response, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+
+	if response == "y" || response == "Y" {
+
+		// loop through accounts list and untag and move them back into root OU
+		for _, id := range accountIdList {
+
+			err = o.untagAccount(id)
+			if err != nil {
+				return err
+			}
+
+			err = o.moveAccount(id, rootID, destinationOU)
+			if err != nil {
+				return err
+			}
+		}
+
+		if accountUsername != "" {
+			// Delete login profile
+			err := o.deleteLoginProfile(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete access keys
+			err = o.deleteAccessKeys(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete signing certificates
+			err = o.deleteSigningCert(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete policies
+			err = o.deletePolicies(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete attached policies
+			err = o.deleteAttachedPolicies(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete groups
+			err = o.deleteGroups(accountUsername)
+			if err != nil {
+				return err
+			}
+			// Delete user
+			err = o.deleteUser(accountUsername)
+			if err != nil {
+				return err
+			}
```

Comment:
I see that we're not deleting roles at all here.  We'll want to also list and delete any roles.  The only role we won't want to delete is the role named: `OrganizationAccountAccessRole`, otherwise we lose access to that account.

I'll leave the decision to add this into v1 of this function to you, otherwise we can card the deletion of the roles and circle back to this at some other point.